lib/schedule.js

```diff
@@ -73,10 +73,42 @@ function cancelJob(job) {
   return success;
 }
 
+function gracefulShutdown() {
+  process.on('SIGINT', function () {
+    const jobs = Object.values(scheduledJobs);
+    jobs.forEach(function (job) {
+      job.cancel();
+    });
+
+    let running = false;
+    for (let i = 0; i < jobs.length; i++) {
+      if (jobs[i].running > 0) {
+        running = true;
+        break;
+      }
+    }
+
+    if (running) {
+      setInterval(function () {
+        for (let i = 0; i < jobs.length; i++) {
+          if (jobs[i].running > 0) {
+            return;
+          }
+        }
+  
+        process.exit(0);
```

Comment:
would be better if it was a graceful shutdown just of node-schedule, not entire Node process

Reply:
This also unlikely to work properly in browser

Reply:
I never thought that this would work in browsers. Is there a good way to just shutdown node-schedule?

Reply:
Can't you just remove the SIGINT wrapping part of the `gracefulShutdown` function and make it do what SIGINT callback is currently doing? That way whoever will invoke `gracefulShutdown()` will effectively get node-schedule stopped after a while. Better yet, you can return a promise that would resolve at the point where you do process.exit(0) now.

Reply:
I thought of terminating the process after processing all the jobs when the system sends an interrupt signal. If you remove the SIGINT wrapping part, you will not be able to handle the interrupt signal event when it occurs.

Reply:
What I mean is people could do
```
process.on('SIGINT', async function () { 
  schedule.gracefulShutdown()
  .then(() => { process.exit(0) })
}
```
when running on node, and just
```
await schedule.gracefulShutdown()
```
on browser, if that's how they want to consume it.

And it probably should be documented.

Reply:
Great Idea! I'll do that.

Reply:
Added a separate example to the document that can gracefully terminate with a SIGINT callback.